tests/run.sh

```diff
@@ -32,13 +59,17 @@ echo "-------- Building"
 RUST_BACKTRACE=1 cargo build
 
 echo "-------- Testing"
+start_swtpm
 mkdir -p /var/lib/keylime
-TCTI=tabrmd:bus_type=session RUST_BACKTRACE=1 RUST_LOG=info \
+RUST_BACKTRACE=1 RUST_LOG=info \
 KEYLIME_CONFIG=$PWD/keylime-agent.conf \
-cargo test --features testing -- --nocapture
+cargo test --features testing -- --nocapture --test-threads=1
+
+# Flush contexts
+tpm2_flushcontext -t -l -s
```

Comment:
Why do we need flushing? We should ideally flush everything in the tests. (Most of the things should be flushed by the tss-esapi abstraction anyway)

Reply:
Empirically I can affirm that they are not flushed, unless explicitly called in the code. Trust me, I spent hours finding the leaks in our code, and modifying the tests to flush the created keys and sessions.

When running the `swtpm` without a resource manager (`tpm2-abrmd`), then it can only hold 3 keys in memory. This number is easily reached if we run tests in parallel, causing errors due to lack of memory. For that reason I had to add the `--test-threads=1` to run the tests one after the next, allowing the keys to be flushed in between.

I modified the `QuoteData::fixture` implementation to create the AK and EK keys only once. The problem is that I did not find a way to cleanup those keys after the tests are executed. So I had to add the `tpm2_flushcontext`.

BTW, I see I forgot to flush the keys at the end of the script, I'll add that.

I also found a deadlock I added by accident. I'll fix this as well.

Reply:
I found a way! But there are still some leakages, I'm still searching the sources

Reply:
In the end I was able to find and cleanup all the points were the tests were leaving leftover key handles and added the cleanup. With this, we don't leave any leftover handle, which makes the `tpm2_flushcontext` there unnecessary. I removed it from the script.